pal/samples/esp8266/telemetry_sample.c.ino

```diff
@@ -0,0 +1,232 @@
+
+static int callbackCounter;
+int receiveContext = 0;
+
+static IOTHUBMESSAGE_DISPOSITION_RESULT receive_message_callback(IOTHUB_MESSAGE_HANDLE message, void* userContextCallback)
+{
+    int* counter = (int*)userContextCallback;
+    const char* buffer;
+    size_t size;
+    MAP_HANDLE mapProperties;
+    const char* messageId;
+
+    // Message properties
+    if ((messageId = IoTHubMessage_GetMessageId(message)) == NULL)
+    {
+        messageId = "<null>";
+    }
+
+    // Message content
+    if (IoTHubMessage_GetByteArray(message, (const unsigned char**)&buffer, &size) != IOTHUB_MESSAGE_OK)
+    {
+        LogInfo("unable to retrieve the message data\r\n");
+    }
+    else
+    {
+        LogInfo("Received Message [%d]\r\n Message ID: %s\r\n Data: <<<%.*s>>> & Size=%d\r\n", *counter, messageId, (int)size, buffer, (int)size);
+        // If we receive the work 'quit' then we stop running
+        if (size == (strlen("quit") * sizeof(char)) && memcmp(buffer, "quit", size) == 0)
+        {
+            g_continueRunning = false;
+        }
+    }
+
+    /* Some device specific action code goes here... */
+    (*counter)++;
+    return IOTHUBMESSAGE_ACCEPTED;
+}
+
+
+static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback)
+{
+    (void)userContextCallback;
+    // When a message is sent this callback will get envoked
+    g_message_count_send_confirmations++;
+    LogInfo("Confirm Callback");
+//    LogInfo("Confirmation callback received for message %lu with result %s\r\n", (unsigned long)g_message_count_send_confirmations, MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONFIRMATION_RESULT, result));
+}
+
+static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context)
+{
+    (void)reason;
+    (void)user_context;
+    // This sample DOES NOT take into consideration network outages.
+    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
+    {
+        LogInfo("The device client is connected to iothub\r\n");
+    }
+    else
+    {
+        LogInfo("The device client has been disconnected\r\n");
+    }
+}
+
+void setup() {
+    int result = 0;
+ 
+    sample_init(ssid, pass);
+
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, protocol);
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+
+    LogInfo("Creating IoTHub Device handle\r\n");
+    // Create the iothub handle here
+    if (device_ll_handle == NULL)
+    {
+        LogInfo("Failure createing Iothub device.  Hint: Check you connection string.\r\n");
+    }
+    else
+    {
+        // Set any option that are neccessary.
+        // For available options please see the iothub_sdk_options.md documentation
+
+    // turn off diagnostic sampling
+    int diag_off=0;
+    IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_DIAGNOSTIC_SAMPLING_PERCENTAGE, &diag_off);
```

Comment:
lets get this indented following the guidelines (indented for each code block)

Reply:
Done.